pkg/utils/namespace.go

```diff
@@ -0,0 +1,96 @@
+package utils
+
+import (
+	"context"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/klog/v2"
+	addonapiv1alpha1 "open-cluster-management.io/api/addon/v1alpha1"
+)
+
+func AgentInstallNamespaceFromDeploymentConfig(
+	adcgetter func(ctx context.Context, namespace, name string) (*addonapiv1alpha1.AddOnDeploymentConfig, error),
```

Comment:
think this one is supposed to be used in a controller to update status on addon?

How about we have a func to sync namespace in config to addon.status?